packages/kit/test/utils.js

```diff
@@ -279,42 +280,85 @@ if (!test_browser_device) {
 	);
 }
 
-/** @type {import('@playwright/test').PlaywrightTestConfig} */
-export const config = {
-	forbidOnly: !!process.env.CI,
-	// generous timeouts on CI
-	timeout: process.env.CI ? 45000 : 15000,
-	webServer: {
-		command: process.env.DEV ? 'pnpm dev --force' : 'pnpm build && pnpm preview',
-		port: process.env.DEV ? 5173 : 4173
+export const app_ports = /** @type {const} */ {
```

Comment:
I wanted to figure out how to assign these fully-dynamically at runtime, but didn't have any ideas that sounded very good